driver/postgres/driver.go

```diff
@@ -98,15 +98,27 @@ func (db *postgres) ApplyMigrations(ctx context.Context, files []file.File, up b
 
 	for _, file := range files {
 		if _, err := tx.ExecContext(ctx, file.SQL); err != nil {
+			if rollbackErr := tx.Rollback(); rollbackErr != nil {
```

Comment:
I actually liked you previous solution (in SSO) with defining separate rollback function to not duplicate the code. Can you please implement the same pattern here?